src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler.cs

```diff
@@ -41,6 +41,40 @@ protected override void NextAction(IEditorOperations editorOperation, Action nex
             editorOperation.InsertNewLine();
         }
 
+        protected override bool TreatAsReturn(Document document, int position, CancellationToken cancellationToken)
+        {
+            var root = document.GetSyntaxRootAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+
+            var endToken = root.FindToken(position);
```

Comment:
Personally, I found these names confusing. Maybe "possibleOpenBrace" and "possibleCloseBrace", etc. I understand that you mean "start" and "end" in the typical Formatting Span sense, but I was confused in this context. Maybe get other opinions.